multicluster/controllers/multicluster/clusterclaim_conversion_controller.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2023 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicluster
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mcsv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	mcsv1alpha2 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha2"
+)
+
+// Since Antrea 1.13, ClusterClaim CRD will be removed from Antrea Multi-cluster API.
+// ClusterClaimConversionController is responsible to convert existing ClusterClaims
+// into clusterID and clusterSetID of ClusterSet for smooth upgrade without manual
+// intervention.
+type ClusterClaimConversionController struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	namespace string
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewClusterClaimConversionController(
+	Client client.Client,
+	Scheme *runtime.Scheme,
+	namespace string,
+) *ClusterClaimConversionController {
+	controller := &ClusterClaimConversionController{
+		Client:    Client,
+		Scheme:    Scheme,
+		namespace: namespace,
+		queue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ClusterClaimConversionController"),
+	}
+	return controller
+}
+
+// Enqueue will be called after ClusterClaimConversionController is initialized.
+func (c *ClusterClaimConversionController) Enqueue() {
+	// The key can be anything as we only have single item.
+	c.queue.Add("key")
+}
+
+// Run starts the ClusterClaimConversionController and blocks until stopCh is closed.
+// it will run only once if no error happens.
+func (c *ClusterClaimConversionController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting ClusterClaimConversionController")
+	defer klog.InfoS("Shutting down ClusterClaimConversionController")
+
+	if err := c.RunOnce(); err != nil {
```

Comment:
Why we just run once? Should not we watch ClusterClaims and ClusterSets and convert at new changes when necessary?

BTW, I am totally fine to do conversion in another PR, or even skip it if we do not know any production user.

Reply:
OK, I will remove the conversion controller, maybe we can add it in the future if it's necessary.

Reply:
Removed conversion controller related codes.